### DISCUSSION TITLE
glTF Working Group Meeting 2022-01-06

### DISCUSSION THREAD

**Discussion by antpb** [omigroup/gltf-extensions#47] [General]
### Agenda

- Discuss Explainer doc: https://github.com/omigroup/gltf-extensions/pull/46
- Godot Doc
- Blender Addon Updates
- Collider extension
- Animation retargeting for VRM workflow

**Comment by fire** [comment]
Work for gltf2 buffers and extensions is now merged in Godot Engine as of https://github.com/godotengine/godot/commit/393a44b275152c660b2010598661b2730c9c60a8

![image](https://user-images.githubusercontent.com/32321/148462349-dac8569a-685a-48bd-8a27-b68545765066.png)

![editor_screenshot_2022-01-06T144044](https://user-images.githubusercontent.com/32321/148462446-e39a601a-eef8-41ca-9383-8dcaaf02720b.png)

Existing and pending work for glTF2 extensions https://github.com/godotengine/godot-proposals/issues/3305:

```
virtual Dictionary get_import_settings() { return import_settings; }
virtual Dictionary get_export_settings() { return export_settings; }
virtual void set_import_settings(Dictionary p_settings) { import_settings = p_settings; }
virtual void set_export_settings(Dictionary p_settings) { export_settings = p_settings; }
virtual Error import_preflight(Ref<GLTFDocument> p_document, Dictionary p_export_settings) { return OK; }
virtual Error import_post(Ref<GLTFDocument> p_document, Node *p_node, Dictionary p_export_settings) { return OK; }
virtual Error export_preflight(Ref<GLTFDocument> p_document, Node *p_node, Dictionary p_import_settings) { return OK; }
virtual Error export_post(Ref<GLTFDocument> p_document, Dictionary p_import_settings) { return OK; }
// Error import_animation(gltf2_animation, Ref<Animation>, Node *,Dictionary p_export_settings) { return OK; }
// Error import_animation_channel(gltf2_animation_channel, channels, Node *, bake_bone, bake_channel, bake_range_start, import_bake_range_end, action_name,Dictionary p_export_settings) {}
// Error import_animation_channel_target(gltf2_animation_channel_target, channels, Node *, bake_bone, bake_channel,Dictionary p_export_settings) {}
// Error import_animation_sampler(gltf2_sampler, Node *, channels, bake_bone, bake_channel, bake_range_start, bake_range_end, action_name,Dictionary p_export_settings) {}
// Error import_asset(gltf2_asset, p_export_settings) {}
// Error import_camera(gltf2_camera, Camera3D *, p_export_settings) {}
// Error import_gltf(gltf2_plan, p_export_settings) {}
// Error import_image(gltf2_image, p_export_settings) {}
// Error import_joint(gltf2_node, Skeleton3D *, BoneId, p_export_settings) {}
// Error import_material(gltf2_material, Ref<Material>, p_export_settings) {}
// Error import_material_pbr_metallic_roughness(gltf2_material, Ref<BaseMaterial3D>, orm_texture, p_export_settings) {}
// Error import_material_unlit(gltf2_material, Ref<Material>, p_export_settings) {}
// Error import_mesh(gltf2_mesh,Ref<ArrayMesh>, vertex_groups, modifiers, skip_filter, material_names, p_export_settings) {}
// Error import_sampler(self, gltf2_sampler, Ref<Material>, p_export_settings) {}
// Error import_node(gltf2_node, Node *, p_export_settings) {}
// Error import_scene(gltf2_scene, Node *, p_export_settings) {}
// Error import_skin(gltf2_skin,  Ref<Skin>, p_export_settings) {}
// Error import_texture(gltf2_texture, p_export_settings) {}
// Error import_texture_info(gltf2_texture_info, p_export_settings) {}
// Error import_merge_animation_extensions(gltf2_animation_source, get_gltf2_animation_destination, p_export_settings)
```

**Comment by antpb** [comment]
Note: Field trip to Third Room soon!

**Comment by antpb** [comment]
3DWebWorldz.com to help in validation of audio emitter extension. thank you! @antpb to assist in implementation

**Comment by RangerMauve** [comment]
https://docs.godotengine.org/en/stable/getting_started/workflow/assets/importing_scenes.html#import-hints

**Comment by RangerMauve** [comment]
I'm gonna lead putting together a proposal for the collission discussion. Going to schedule a co-working sesh for early next week with anyone that's interesed.

I think to start, collecting links to prior art and outlining what we want to support for the first version would be helpful.

**Comment by pathorn** [comment]
re: Colliders:
Godot supports generating navmesh? collisions?

Static body vs dynamic / rigid body. Example of rigid body: basketball? (spawn in an object?)

To handle meshes used only for collision and not in the world, we should have a separate extension property to link to a collider mesh.

We also want to handle primitives, such as box, sphere or capsule. Which primitives should be supported?
meet next week to start drafting

**Comment by fire** [comment]
Wanted to record the idea that we learned from vrm that the format should degrade in non extended standard gltf. VRM didn’t use standard gltf handedness and forward vector.
